keel-intent/src/main/kotlin/com/netflix/spinnaker/keel/intent/processor/SecurityGroupIntentProcessor.kt

```diff
@@ -103,9 +108,37 @@ class SecurityGroupIntentProcessor
     throw NotImplementedError("Only amazon security groups are supported at the moment")
   }
 
-  private fun currentStateUpToDate(currentState: Set<SecurityGroup>, desiredState: Set<SecurityGroup>): Boolean {
-    log.warn("Current state update check is not implemented")
-    return false
+  private fun currentStateUpToDate(intentId: String,
+                                   currentState: Set<SecurityGroup>,
+                                   desiredState: SecurityGroupSpec): Boolean {
+    if (currentState.size != desiredState.regions.size) return false
+
+    val desired = securityGroupConverter.convertToState(desiredState)
+
+    val statePairs = mutableListOf<Pair<SecurityGroup, SecurityGroup>>()
+    desired.forEach { d ->
+      val key = "${d.accountName}/${d.region}/${d.name}"
+      currentState.forEach { c ->
+        if ("${c.accountName}/${c.region}/${c.name}" == key) {
+          statePairs.add(Pair(c, d))
+        }
+      }
+    }
```

Comment:
Determining whether or not the system needs to be updated for security groups is a little more involved since the relationship is `1 Intent -> n [system resource]`, so we have to create state pairs based on a key namespace. Something to keep in mind for other composite intents, like load balancers.